br/pkg/lightning/errormanager/resolveconflict_test.go

```diff
@@ -0,0 +1,798 @@
+		WithArgs(0, "a", nil, nil, data6NonclusteredKey, data6NonclusteredValue, 1).
+		WillReturnResult(driver.ResultNoRows)
+	mockDB.ExpectCommit()
+	mockDB.ExpectQuery("\\QSELECT raw_key, raw_value FROM `lightning_task_info`.conflict_error_v2 WHERE table_name = ? AND is_data_kv = 1 ORDER BY raw_key\\E").
+		WillReturnRows(sqlmock.NewRows([]string{"raw_key", "raw_value"}).
+			AddRow(data2NonclusteredKey, data2NonclusteredValue).
+			AddRow(data6NonclusteredKey, data6NonclusteredValue))
+
+	cfg := config.NewConfig()
+	cfg.TikvImporter.DuplicateResolution = config.DupeResAlgReplace
+	cfg.App.TaskInfoSchemaName = "lightning_task_info"
+	em := errormanager.New(db, cfg, log.L())
+	err = em.Init(ctx)
+	require.NoError(t, err)
+
+	fnGetLatestCount := atomic.NewInt32(0)
+	fnDeleteKeyCount := atomic.NewInt32(0)
+	pool := utils.NewWorkerPool(16, "resolve duplicate rows by replace")
+	err = em.ReplaceConflictKeys(
+		ctx, tbl, "a", pool,
+		func(ctx context.Context, key []byte) ([]byte, error) {
+			fnGetLatestCount.Add(1)
+			switch {
+			case bytes.Equal(key, data2RowKey):
+				return data2RowValue, nil
+			case bytes.Equal(key, data2NonclusteredKey):
+				if fnGetLatestCount.String() == "3" {
+					return data2NonclusteredValue, nil
+				} else {
+					return nil, tikverr.ErrNotExist
+				}
+			case bytes.Equal(key, data6RowKey):
+				return data6RowValue, nil
+			case bytes.Equal(key, data6NonclusteredKey):
+				if fnGetLatestCount.String() == "6" {
+					return data6NonclusteredValue, nil
+				} else {
+					return nil, tikverr.ErrNotExist
+				}
+			case bytes.Equal(key, data2IndexKey):
+				return data2IndexValue, nil
+			case bytes.Equal(key, data3RowKey):
+				return data3RowValue, nil
+			case bytes.Equal(key, data6IndexKey):
+				return data3RowValue, nil
+			case bytes.Equal(key, data7RowKey):
+				return data7RowValue, nil
+			default:
+				return nil, fmt.Errorf("key %v is not expected", key)
+			}
+		},
+		func(ctx context.Context, key []byte) error {
+			fnDeleteKeyCount.Add(1)
+			if !bytes.Equal(key, data2NonclusteredKey) && !bytes.Equal(key, data6NonclusteredKey) && !bytes.Equal(key, data2IndexKey) && !bytes.Equal(key, data3RowKey) && !bytes.Equal(key, data6IndexKey) && !bytes.Equal(key, data7RowKey) {
+				return fmt.Errorf("key %v is not expected", key)
+			}
+			return nil
+		},
+	)
+	require.NoError(t, err)
+	require.Equal(t, int32(16), fnGetLatestCount.Load())
+	require.Equal(t, int32(6), fnDeleteKeyCount.Load())
+	err = mockDB.ExpectationsWereMet()
+	require.NoError(t, err)
+}
+
+func TestReplaceConflictOneKeyNonclusteredPk(t *testing.T) {
+	p := parser.New()
+	node, _, err := p.ParseSQL("create table a (a int primary key nonclustered, b int not null, c text, key key_b(b));")
+	require.NoError(t, err)
+	mockSctx := mock.NewContext()
+	mockSctx.GetSessionVars().EnableClusteredIndex = variable.ClusteredIndexDefModeOff
+	info, err := ddl.MockTableInfo(mockSctx, node[0].(*ast.CreateTableStmt), 1)
```

Comment:
```suggestion
	info, err := ddl.MockTableInfo(mockSctx, node[0].(*ast.CreateTableStmt), 108)
```

and we can delete line 261.

Also for other tests